docs/design.md

```diff
@@ -0,0 +1,128 @@
+# Overview
+
+Concord-bft is an implementation of the [SBFT
+Protocol](https://arxiv.org/abs/1804.01626). SBFT implements State Machine
+Replication (SMR) in the byzantine model. It is a generic protocol, with
+pluggable interfaces for communication, storage, and state transfer. This
+document covers the implementation of concord-bft. When the implementation
+differs from the paper, it will be mentioned explicitly.
+
+It is important to note that this document covers the code as it exists **NOW**.
+When the code changes substantially this document should be updated.
+
+# Public Interfaces
+
+Concord-bft is a library meant to be used by application code. The external user
+API for the core engine is described by a [set of structs and abstract
+interfaces](../bftengine/include/bftengine). The file names are pretty self
+explanatory. Important structures will be mentioned explicitly below.
+
+All user interfaces live in the `bftEngine` namespace.
+
+# Clients
+A default client,
+[SimpleClient](../bftengine/include/bftengine/SimpleClient.hpp), is included
+with concord-bft. It is just one manner of implementing a client, and is
+implemented in C++ utilizing the `ICommunication` and `IReceiver` interfaces
+below. Additional clients can be implemented by utilizing the ClientMessages
```

Comment:
I don't see the string ClientMessage in the code base. Should this be two words?

Reply:
Yes. Good catch!